test/src/lib.rs

```diff
@@ -104,15 +110,32 @@ impl Rpc {
 
 		// extract interesting part from the response
 		let extracted = match serde_json::from_str(&response).expect("We will always get a single output.") {
-			response::Output::Success(response::Success { result, .. }) => serde_json::to_string_pretty(&result),
-			response::Output::Failure(response::Failure { error, .. }) => serde_json::to_string_pretty(&error),
+			response::Output::Success(response::Success { result, .. }) => success_formatter(&result),
+			response::Output::Failure(response::Failure { error, .. }) => error_formatter(&error),
 		}.expect("Serialization is infallible; qed");
 
-
 		println!("\n{}\n --> {}\n", request, extracted);
 
 		extracted
 	}
+
+	/// Perform a single, synchronous method call and return pretty-printed value
+	pub fn request<T>(&self, method: &str, params: &T) -> String where
+		T: serde::Serialize,
+	{
+		let success_formatter = serde_json::to_string_pretty;
+		let error_formatter = serde_json::to_string_pretty;
+		self.request_helper(method, params, success_formatter, error_formatter)
+	}
+
+	/// Perform a single, synchronous method call.
+	pub fn make_request<T>(&self, method: &str, params: &T) -> String where
```

Comment:
I don't really like the name that much, but don't have a single better suggestion. Maybe:
1. `request_compact(`?
2. `make_request(&str, &T, Encoding::Compact)` - `enum Encoding` would decide on the type of encoding.